shared/ui/core/src/main/java/com/ivy/ui/ComposeViewModel.kt

```diff
@@ -22,4 +24,12 @@ abstract class ComposeViewModel<UiState, UiEvent> : ViewModel() {
      * in the UI to be processed in the ViewModel.
      */
     abstract fun onEvent(event: UiEvent)
+
+    private val observableEvents = MutableLiveData<ViewModelEvent>()
```

Comment:
Oh shit 😬 Revert this - the Compose ViewModel already supports proper event handling. Read the docs about UI layer and screen architecture in Ivy Wallet's dev guidelines

Reply:
From what i can learn is that the event flow is between view model and corresponding composable. I want to flow a event to the main activity to launch an intent and i don't want to put context reference in the view model.

Reply:
Okay, putting stuff in MainActivity is a terrible idea because:
- it breaks our modularization strategy 
- creates the RootActivity.kt monolith that I want to break
- increase complexity

For your case, a good solution can be:
 1. Create `AppSettingsNavigator` class
 2. Inject `@ApplicationContext context: Context`
 3. Implement a `fun openLanguageSettings()`
 4. Inject this Navigator in the SettingsViewModel
 5. Enjoy!

Also, injecting an application context in the VM isn't an issue. @suyash01 wdyt?

Reply:
Ok, let me try the solution you are proposing.

I don't want context in VM is due to the fact VM will have shorter lifecycle than the activity and it also makes it difficult to write test cases later on and also decoupling reasons.

Reply:
A few points on this:
- You can write tests by `mockk<Context>()`

The app context lives as long as the app lives. What's the drawback from having the app's context in the VM?
- VM can still have shorter lifecycle than the context and will be successfully garbage collected. 
- There's no case where the VM will outlive the app context => causing a memory leak for the app context

That's a common misconception in the Android community. Also, we don't use the app context directly in the VM but through some proxy class and DI. If we happen to migrate to KMP (one good reason for not having the Context) we can just make the Navigator class an interface, thus abstract it